Posts/2021/03/DarwinJS-Contribution-LightningFastandEasyProvisioningofGitwithSSHKeyAuthenticationonWindows.md

```diff
@@ -0,0 +1,117 @@
+---
+post_title: Lightning Fast and Easy Provisioning of Git with SSH Key Authentication on Windows
+username: DarwinJS@sanoys.com
+Catagories: PowerShell
+tags: Git, DevOps
+Summary: Getting your team setup with Git over SSH as quickly as possible!
+featured_image: ./media/DarwinJS-Contribution-LightningFastandEasyProvisioningofGitwithSSHKeyAuthenticationonWindows/windows-git-ssh.png
+CanonicalURL: https://missionimpossiblecode.io/post/lightning-fast-and-easy-provisioning-of-git-with-ssh-key-authentication-on-windows/
+---
+
+
+Maybe you have a team of Windows developers that are onboarding for your new Git server installation or maybe you've decided to drop http password authentication to your existing Git server (due to it's many problems).  Your next steps may well be into a rough and rocky rabbit hole when you were holding out hope for simplicity (you know the kind you've fallen into before if you've been in tech for more than about 45 minutes).
+
+The guides on the internet for getting Windows setup for SSH authentication for Git are unnecessarily complex.
+
+My inner tool smith really loathes when the very first steps into something new are fraught with rocky rabbit holes - so I took on the challenge of creating an easier way. 
+
+The resultant tool is a 20 line PowerShell script that deploys Git, configures SSH and leaves the public key on your clipboard so you can paste it into GitLab or any other Git collaborative webserver. There is also an optional connectivity test.
+
+<!-- more -->
+
+## Reasons For Moving to SSH
+
+There are multiple reasons you may want to move your Windows developers to SSH authentication for Git:
+
+1. You want to get away from git storing local passwords - whether in the git config or in Windows Credentials (with the windows credential helper) because it is pure pain to walk people through how to find and update this password when they change it on the Git server.
+2. You want to avoid both http passwords and the http protocol for git.
+
+## Conventional Wisdom on SSH Configuration
+
+The conventional wisdom solution offers many steps that are roughly:
+
+1. Installing git manually.
+2. Installing the well known Windows SSH client Putty
+3. Installing Putty's key generator.
+4. Converting the non-compatible putty generated key into an ssh compatible one.
+5. Precisely placing the SSH key on disk.
+6. Precisely and ~~manually~~ permissioning the SSH key and it's parent folder (ssh is purposely fussy about this in order to keep the key secure).
```

Comment:
Did you intend to use strikethrough formatting here? It was not in the original blog post. Is you want to use it then I suggest the following change.

```suggestion
6. Precisely ~~and manually~~ permissioning the SSH key and it's parent folder (ssh is purposely fussy about this in order to keep the key secure).
```

Reply:
This was a mistake - fixed in another commit.